*** snipMate.vim.old	2009-04-23 02:30:12.703125000 +0900
--- snipMate.vim	2009-04-23 01:41:56.890625000 +0900
***************
*** 187,190 ****
--- 187,202 ----
  	let num = inputlist(snippet) - 1
  	return num == -1 ? '' : s:multi_snips[a:scope][a:trigger][num][1]
  endf
+ 
+ fun GetSnippetsList(ft)
+ 	let s = {}
+ 	for i in ['s:snippets["_"]', 's:multi_snips["_"]',
+ 		\ 's:snippets[a:ft]', 's:multi_snips[a:ft]']
+ 		if exists(i)
+ 			let s = extend(s, deepcopy(eval(i)))
+ 		endif
+ 	endfor
+ 	return s
+ endf
+ 
  " vim:noet:sw=4:ts=4:ft=vim